### PR TITLE
Prime for Push in Settings

### PR DESCRIPTION
**Pull request by nheagy**
Refs #9185

This adds a button to the push notification settings if the user ignored the prime for push alert after login.

prime for push in settings

To test:
- with a fresh install login to a wpcom account
- when the push notification primer appears, tap "Not now"
- go to Me > Notification Settings > Comments on Other Sites > Push Notifications
- ensure that the new button appears
- repeat for the other two scenarios: tapping "Allow notifications" on the alert after login, and then both allowing and denying push notifications

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift [hunk 1]
```diff
@@ -117,7 +118,13 @@ class NotificationSettingDetailsViewController: UITableViewController {
     }
 
     @IBAction func reloadTable() {
-        sections = isDeviceStreamDisabled() ? sectionsForDisabledDeviceStream() : sectionsForSettings(settings!, stream: stream!)
+        if isDeviceStreamDisabled() {
+            sections = sectionsForDisabledDeviceStream()
+        } else if isDeviceStreamUnknown() {
+            sections = sectionsForUnkownDeviceStream()
+        } else {
+            sections = sectionsForSettings(settings!, stream: stream!)
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift]: Can we unwrap these `!`s?

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift [hunk 2]
```diff
@@ -169,6 +176,20 @@ class NotificationSettingDetailsViewController: UITableViewController {
         return [section]
     }
 
+    private func sectionsForUnkownDeviceStream() -> [Section] {
+        defer {
+            WPAnalytics.track(.pushNotificationPrimerSeen, withProperties: [Analytics.locationKey: Analytics.alertKey])
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift]: Is there a reason this is in a `defer`?

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift [hunk 3]
```diff
@@ -169,6 +176,20 @@ class NotificationSettingDetailsViewController: UITableViewController {
         return [section]
     }
 
+    private func sectionsForUnkownDeviceStream() -> [Section] {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift]: Nitpick, typo: `Unkown`

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift [hunk 4]
```diff
@@ -241,17 +264,30 @@ class NotificationSettingDetailsViewController: UITableViewController {
...
+    private func isDeviceStreamUnknown() -> Bool {
+        return stream?.kind == .Device && pushNotificationsAuthorized == .notDetermined
     }
 
     private func openApplicationSettings() {
         let targetURL = URL(string: UIApplicationOpenSettingsURLString)
         UIApplication.shared.open(targetURL!)
     }
 
+    private func requestNoficationAuthorization() {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift]: Typo: `Nofication`

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift [hunk 5]
```diff
@@ -241,17 +264,30 @@ class NotificationSettingDetailsViewController: UITableViewController {
...
+        defer {
+            WPAnalytics.track(.pushNotificationPrimerAllowTapped, withProperties: [Analytics.locationKey: Analytics.alertKey])
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift]: Same `defer` question as above?

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift (hunk 2 above)
Comment (nheagy) [reply WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift]: I like to defer analytics calls _just in case_. I can remove it if you think it looks too weird 😉

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift (hunk 1 above)
Comment (nheagy) [reply WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift]: Sure—I always like the opportunity to remove some `!` from old code! 👍

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift (hunk 2 above)
Comment (frosty) [reply WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift]: I don't really mind that much, I just wondered if it was really necessary :)

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift (hunk 1 above)
Comment (frosty) [reply WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift]: ! 💥

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift (hunk 2 above)
Comment (nheagy) [reply WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift]: Ok then I left them 😜

**Comment by nheagy** [comment]
@frosty fixed! Except the `defer`s. I could be convinced to remove them too, if they make the code to muddy.

**Comment by Suil3File** [comment]
2fb5f61

**Pull request merged**